lua/nvim-treesitter/ts_utils.lua

```diff
@@ -175,6 +175,29 @@ function M.highlight_node(node, buf, hl_namespace, hl_group)
   M.highlight_range({ node:range() }, buf, hl_namespace, hl_group)
 end
 
+--- Get a compatible vim range (1 index based) from a TS node range.
+--
+-- TS nodes start with 0 and the end col is ending exclusive.
+-- They also treat a EOF/EOL char as a char ending in the first
+-- col of the next row.
+function M.get_vim_range(range, buf)
```

Comment:
This function was long needed. Does it require a special treatment of 0-sized ranges? I probably not since they can't be represented with vim ranges and would not be useful when using this function. Thinking of the range 0,0,0,0. But I don't think there's a possibility that 0-char ranges get passed to that function.

Reply:
Hmm, yeah, well, let's see if this ever happens :laughing: